heisenbridge/__main__.py

```diff
@@ -613,6 +619,10 @@ async def run(self, listen_address, listen_port, homeserver_url, owner, safe_mod
                 else:
                     await self.leave_room(room_id, joined.keys())
 
+        if not self.hidden_room_id:
```

Comment:
I think we should test if the HS supports v9 before even doing this, we can then just test if `hidden_room_id` is set everywhere to know if it exists and is supported.